### DISCUSSION TITLE
How to pass Custom header such as x-tenant-id to Elsa Studio

### DISCUSSION THREAD

**Discussion by shahidriaz** [elsa-workflows/elsa-studio#299] [Q&A]
Dear all,
I am using Elsa 3.3.
I have added a code where I extract a key named "x-tenant-id" from the http context. This then get saved in the WorkflowDefination table.
It is working fine, but now I am lost that how can I pass this x-tenant-id in each request 
e.g. When I click on **Create Workflow** , a workflow is created, but I want to pass the x-tenant-id in the httprequest that is generated by the Create Workflow button.

**Comment by hpposch** [comment]
Hi @shahidriaz,
did you manage to login using multitenantcy?
If yes, could you explain how it works and what i need to get it running.

I really want to try Elsa's multitenantcy feature, but i am not sure on how to get it up and running.

Thanks, 
Heinz Peter

**Reply by hpposch** [reply]
I just got the following setup running:
Elsa.Server.Web from elsa-core
Elsa.Studio.Web from elsa-studio

I get a response from https://localhost:5001/ and also from https://tenant-1.localhost:5001/
Therefore it works.

I am now wondering if elsa-studio already supports multitenancy, and if yes how does that work.
I am trying with http://localhost:5010/ and http://tenant-1.localhost:5010/
But it always shows me the same workflows, it seems that it does not switch the backend for each tenant automatically.
@sfmskywalker can you confirm this behavior?

**Reply by sfmskywalker** [reply]
Hi @hpposch , Elsa Studio itself is tenant-unaware. It only sends authenticated HTTP requests to the backend. It is then up to the backend to establish the correct tenant scope. For example, the access token establishes a ClaimsPrincipal. The tenant resolver components can then use information from the established security context what the current tenant must be.
